content/en/get-started/cast (smart contract).md

```diff
@@ -0,0 +1,116 @@
+
+interface IDSA {
+    function cast(
+        string[] calldata _targetNames,
+        bytes[] calldata _datas,
+        address _origin
+    ) external payable returns (bytes32);
+}
+
+contract InteractingDSA {
+
+    IInstaIndex instaIndex = IInstaIndex(0x2971adfa57b20e5a416ae5a708a8655a9c74f723); // this address is only of mainnet.
+
+    function buildAndCast(address _owner) external {
+        // creating an account
+        address _account = instaIndex.build(_owner, 2, address(0)); // 2 is the most recent DSA version
+        
+        // encoding data to run multiple things through cast on account
+        // Depositing in DSA and then deposit in Compound through DSA.
+        string[] memory _targets = new string[](2);
+        bytes[] memory _data = new bytes[](2);
+        
+        _targets[0] = "BASIC-A";
+        _targets[1] = "COMPOUND-A";
+        
+        bytes4 private constant basicDeposit = bytes4(keccak256("deposit(address,uint256,uint256,uint256)"));
+        bytes4 private constant compoundDeposit = bytes4(keccak256("deposit(string,uint256,uint256,uint256)"));
+        
+        address dai = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+        uint amtToDeposit = 1e18; // 1 DAI
+        
+        _data[0] = abi.encodeWithSelector(basicDeposit, dai, amtToDeposit, 0, 0);
+        _data[1] = abi.encodeWithSelector(compoundDeposit, dai, amtToDeposit, 0, 0);
+        
+        IDSA(_account).cast(_targets, _data, address(0)); // Magic!!
+    }
+
+}   
+```
+Now we can start interacting with the DSA contract.
+Go to Networks to find each chain-related address.
+
+> Go to [Networks](https://docs.instadapp.io/networks/mainnet) to find each chain-related address.
+
+### Usage
+
+* [InstaIndex:](https://github.com/Instadapp/dsa-contracts/blob/master/contracts/registry/index.sol)
+This is the Main Contract for all the Defi Smart Accounts. Used to create a new Defi Smart Account for a user and run a cast function in the new smart account.
+
+
+* [InstaImplementationM1:](https://github.com/Instadapp/dsa-contracts/blob/master/contracts/v2/accounts/module1/Implementation_m1.sol)
+This contract contains most core functions of smart account name cast(). It is only called by owners of smart accounts and has full-fledge access over the smart account. Used also to access all the DSA.
+
+* [build(owner, accountVersion, _origin):](https://github.com/Instadapp/dsa-contracts/blob/master/contracts/registry/index.sol#L170)
+Create a DSA Account using this function. It returns the address of the DSA account created.
+
+```
+owner: Owner of the Smart Account
+accountVersion: Account Module version
+_origin: Where Smart Account is created.
+```
+
+* [cast(_targets, _datas, _origin):](https://github.com/Instadapp/dsa-contracts/blob/master/contracts/v2/accounts/module1/Implementation_m1.sol#L80)
+Using cast() user can access the connectors, which allows the smart account to interact with protocols or set up any settings on the smart account.
+
+```
```

Comment:
Better to put this parameter in Table format.
like this :
![image](https://user-images.githubusercontent.com/63545809/141608154-8360719c-6c98-4485-8ac9-b5a3e37d18d9.png)

Reply:
Done